### ISSUE TITLE
Datapoint submission should generate data zip file

### ISSUE THREAD

**Issue by valllllll2000** [akvo/akvo-flow-mobile#1090]
Once the user presses submit, the zip file will be generated. If that part fails, he will be shown an error SnackBar and the datapoint will remain in "saved" state and we do not leave the form screen.
-> **Error message TBD**
While data zip is being generated we show a "loading" UI item. 
File generation and submitting process should happen on background thread.
! What if user leaves the screen: submission should proceed correctly, once user comes back, we check if submitted, then we reload view to not allow edition.

**Comment by valllllll2000** [comment]
## Test plan
Create survey with signature, image, video, caddisfly e coli tests (needs image) .... 
Place device in flight mode. Create datapoint and submit. Make sure you see a loading after submitting.
Go to settings and publish data. Make sure the data and images are all correct in "published" folder. Upload to dashboard as usual and check that everything is correct.
Enable wifi.
Create a datapoint. Once synced, verify all data is correct in the dashboard. Also check files like images, video or caddisfly images.

**Comment by janagombitova** [comment]
@valllllll2000 I created a survey with all questions also Caddisfly. Collected the data offline and then hit publish. I copied over the entire published folder to my computer and compressed it. Here is the file: https://drive.google.com/open?id=13xIIJOU153RSwArJlpHj27MH7L2vamM2 I imported it via bulk upload and then checked the data in Inspect data tab. 

Survey form on uat1: 
_1.9.36 and 2.6.0 > 1.9.36 all questions > New form 

The two data points with Caddisfly data are called 'caddis'.

![screen shot 2018-08-21 at 13 34 39](https://user-images.githubusercontent.com/12456965/44399228-07632800-a547-11e8-9c41-bd2e3117c8ef.png)
